### PR TITLE
report: handle non-numeric numericValues in calc link

### PR DESCRIPTION
**Pull request by paulirish**
Whelp.. in LR, our undefined's become null and we end up throwing.
image

definitely shoulda gone with this version of the typeof check to begin with. :/

this is a 6.0.1 fix. i imagine we've already hit it in PSI a couple times

### CODE REVIEW

**Comment by brendankenny** [comment]
> our undefined's become null and we end up throwing

OK, something is wrong, then. This really shouldn't happen and we should fix.

> DoubleValue will work well for us. 0 comes through, and then any audit with it not set will come back with no numericValue in JSON, which works just fine for us.

https://github.com/GoogleChrome/lighthouse/pull/8385#discussion_r276814636

**Comment by brendankenny** [comment]
> OK, something is wrong, then. This really shouldn't happen and we should fix.

Yeah, something did break. The CI version of the proto test isn't seeing this problem (both missing `numericValue` and `details` are dropped as expected instead of made `null`). See this purposefully failed `sample_v2` roundtrip for what it looks like there: https://github.com/GoogleChrome/lighthouse/runs/722125320#step:14:596 ([test branch](https://github.com/GoogleChrome/lighthouse/tree/protonull))

@exterkamp is is possible the PSI proto handling is diverging from the library we're using on the null/default value front?

**Comment by paulirish** [comment]
- @paulirish to make a repro.
- @exterkamp to investigate the default value of DoubleValue and why numericValue is giving us a `null`

**Comment by connorjclark** [comment]
Either way, this change is a net positive, can we merge?

File: lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js [hunk 1]
```diff
@@ -288,14 +288,23 @@ describe('PerfCategoryRenderer', () => {
...
       lcp.result.numericValue = undefined;
       lcp.result.score = null;
       const href = renderer._getScoringCalculatorHref(categoryClone.auditRefs);
       expect(href).toContain('largest-contentful-paint=null');
     });
+
+    it('uses null if the metric\'s value is null (LR)', () => {
+      const categoryClone = JSON.parse(JSON.stringify(category));
+      const lcp = categoryClone.auditRefs.find(audit => audit.id === 'largest-contentful-paint');
+      lcp.result.numericValue = null;  // In LR, undefined becomes null
```
Comment (patrickhulce) [inline lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js:303]:
```suggestion
      lcp.result.numericValue = null; // In LR, undefined becomes null
```

Comment (brendankenny) [reply lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js:303]: the comment is still wrong even without the extra space, though :D

Comment (patrickhulce) [reply lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js:303]: I thought the status was it *should* be wrong, but for some reason we're seeing it be true?

Comment (brendankenny) [reply lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js:303]: I think it was that only Paul (possibly?) saw it, and only in the non-real deployment

Comment (patrickhulce) [reply lighthouse-core/test/report/html/renderer/performance-category-renderer-test.js:303]: we think it *should* be impossible, but this is a better way to handle this situation even if it's not possible, so we're proceeding here.

**Review by connorjclark — APPROVED** [review]

File: lighthouse-core/report/html/renderer/performance-category-renderer.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
     if (fmp) v5andv6metrics.push(fmp);
 
     const metricPairs = v5andv6metrics.map(audit => {
-      const value = typeof audit.result.numericValue !== 'undefined' ?
+      const value = typeof audit.result.numericValue === 'number' ?
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/performance-category-renderer.js:120]: this fix went in with https://github.com/GoogleChrome/lighthouse/pull/10954

**Pull request merged**